### ISSUE TITLE
Site header navigation (mobile): unable to tab to sub-menu or press enter to show/hide sub-menu items - ITSADSSD-33883

### ISSUE THREAD

**Issue by jen-lui** [uq-its-ss/design-system#231]
**Unable to navigate to sub-menu in mobile site header navigation**
When using the keyboard to navigate site header navigation, you are unable to tab to the sub-menu chevron and press enter to toggle to show/hide the sub-menu items. You can only tab and navigate to top-level menu items.

**To Reproduce**
1. Inspect kitchen sink and emulate a mobile device
2. Tab through to hamburger icon
3. Press enter to open the menu and attempt to tab to sub-menu chevron

**Expected behaviour**
- A user is able to tab to the sub-menu chevron and press enter to show/hide the sub-menu items. 
- A user is able to tab to a sub-menu item and press enter to navigate to that page
- A user can only have one sub-menu toggled open at a time.

**Questions**
Should the sub-menu close when you tab off it?

**Screenshots**
![Screen Shot 2021-07-29 at 1 28 41 pm](https://user-images.githubusercontent.com/6946929/127428124-7a1e1d7d-c6c4-4718-98c8-10a8864a29ff.png)

**Additional context**
Current implementation on UQ home page has some of the expected behaviour and functionality https://www.uq.edu.au/
